### DISCUSSION TITLE
Liferea future and CTK

### DISCUSSION THREAD

**Discussion by sjehuda** [lwindolf/liferea#1394] [Ideas]
Good day!

In respond to [20 years of liferea and the future](https://lzone.de/liferea/blog/20-years-of-Liferea-and-the-future), which [I have read](https://github.com/lwindolf/liferea/discussions/974#discussioncomment-11090667) last week.

> BTW it will be as ugly and clumsy as many people say the Liferea GUI is, but the functionality will be great :-)

Liferea is not ugly. Liferea is professional, and without nonsense features.

> The different aspects of this are the major GTK releases causing massive refactorings, continuous API deprecations causing compiler noise and work that does not benefit the features.

I probably would advise for KDE or Qt, but when I think more about it, and considering the good interface of Liferea, I would consider FLTK.

[Dillo HTML Browser](https://dillo.org/) is made with FLTK, and there is a [modified version](https://www.toomanyatoms.com/software/mobilized_dillo.html) (Mobilized Dillo) which is for mobile devices, such as Mobian and postmarketOS. @dillo-browser

[Yoshimi](https://yoshimi.github.io/) is also made with FLTK. @yoshimi

Those software are good, and do tasks as expected.

FLTK might be the graphical toolkit that we would want to be utilized for Liferea.

Kindly,
Schimon

**Comment by lwindolf** [comment]
I'm not really convinced of switching to FLTK

**Reply by sjehuda** [reply]
Yes. Apparently FLTK is a toolkit for C++, and Liferea is written in C.

**Comment by sjehuda** [comment]
Provided that a Python plugins be possible, I would be very much in
favour, also due to the lower amount of resources which FLTK requires.

The interface is good, without animations, rounded corners, and other
unnecessary facy effects.

I would not expect the various of disappointing surprises of GTK+,
including the recent decision to remove support for X11/Xorg, which
strengthen the sense that Wayland is expected to be utilized as a
spyware agent.

A couple of good telecommunication (i.e. internet) software that are
based upon the FLTK toolkit:

https://dillo-browser.github.io

http://florb.shugaa.de

**Comment by sc0w** [comment]
or liferea can be migrated to ctk:

https://github.com/cafe-desktop/ctk

it is a gtk3 fork

it is used by CAFE desktop (MATE desktop fork):

https://github.com/cafe-desktop

**Reply by sjehuda** [reply]
This could be good. I have replaced FLTK by CTK.

Reference: https://github.com/amule-project/amule/discussions/403#discussioncomment-12799634

**Comment by sjehuda** [comment]
I want to add, that as Linux rapidly turns into a spyware with the incorporation of SystemD, and Wayland is unjustly imposed on people, and Linux is more attractive to "gamers", then Linux would eventually be as proprietary systems, with careless and senseless people.

However, the audience of Liferea would probably refer to Linux distributions that utilize XLibre, OpenRC (or another init system), and FreeBSD.

Hence, I sense that neglecting GTK+ would eventually be a priority in the future.

* https://github.com/gpodder/gpodder/discussions/1784
* https://forum.artixlinux.org/index.php/topic,8264.0.html
* https://forums.freebsd.org/threads/gtk-to-remove-x11-backend.96670/
* https://gist.github.com/probonopd/9feb7c20257af5dd915e3a9f2d1f2277

**Comment by sjehuda** [comment]
For record.

There is a convenient internet browser which is called Lagrange, which is also written in C, and it utilizes SDL as a graphical toolkit.

> Like Gemini, Lagrange has been designed with minimalism in mind. It depends on a small number of essential libraries. It is written in C and uses SDL for hardware-accelerated graphics. OpenSSL is used for secure communications.

https://gmi.skyjake.fi/lagrange/

I think that SDL is perfect.

**Comment by sjehuda** [comment]
This is an interesting project of @stefan11111

> Compatibility library to run gtk3 apps with gtk2

https://github.com/stefan11111/gtk3-to-gtk2

Via https://delightful.coding.social/delightful-x11/

**Comment by sjehuda** [comment]
Closed in favour of discussion https://github.com/lwindolf/liferea/discussions/1479
